src/cpu.c

```diff
@@ -378,17 +364,127 @@ void OPC_LD_SP(void) {
     cpu.cycle_count += 3;
 }
 
-void OPC_LD_HL_MINUS_A(void) {
-    ++cpu.cycle_count;
-    uint16_t i = CPU_DREG_HL;
-    mmu_write_byte(i, CPU_REG_A);
-    CPU_DREG_HL = i - 1;
-    ++cpu.cycle_count;
-}
-
 /******************************************************
  *** 8-BIT Loads                                    ***
  ******************************************************/
+void OPC_LD_BC_A(void) {
+    mmu_write_byte(CPU_DREG_BC, CPU_REG_A);
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_DE_A(void) {
+    mmu_write_byte(CPU_DREG_DE, CPU_REG_A);
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_HL_PLUS_A(void) {
+    mmu_write_byte(CPU_DREG_HL, CPU_REG_A);
+    ++CPU_DREG_HL;
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_HL_MINUS_A(void) {
+    mmu_write_byte(CPU_DREG_HL, CPU_REG_A);
+    --CPU_DREG_HL;
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_A_BC(void) {
+    LD_REG_REG(&CPU_REG_A, mmu_get_byte(CPU_DREG_BC));
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_A_DE(void) {
+    LD_REG_REG(&CPU_REG_A, mmu_get_byte(CPU_DREG_DE));
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_A_HL_PLUS(void) {
+    LD_REG_REG(&CPU_REG_A, mmu_get_byte(CPU_DREG_HL));
+    ++CPU_DREG_HL;
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_A_HL_MINUS(void) {
+    LD_REG_REG(&CPU_REG_A, mmu_get_byte(CPU_DREG_HL));
+    --CPU_DREG_HL;
+
+    cpu.cycle_count += 8;
+    ++cpu.PC;
+}
+
+void OPC_LD_FF00a8_A(void) {
+    uint16_t intermediate = 0xFF00 + mmu_get_byte(cpu.PC + 1);
+    mmu_write_byte(intermediate, CPU_REG_A);
+
+    cpu.cycle_count += 12;
+    cpu.PC += 2;
+}
+
+void OPC_LD_A_FF00a8(void) {
```

Comment:
Is there a reason for naming this a8 instead of d8? 🤔

Reply:
Good question, I simply took the naming convention from our  [CPU reference](https://www.pastraiser.com/cpu/gameboy/gameboy_opcodes.html) :D